### ISSUE TITLE
Support for mailing APIs, not just SMTP

### ISSUE THREAD

**Issue by LukeHallett** [FusionAuth/fusionauth-issues#766]
### Problem
FusionAuth only supports SMTP, which can be an issue on some hosting providers, as it may be blocked or limited.

### Solution
Add support for mailing APIs, such as MailGun or SendGrid.

### Alternatives/workarounds
I've had to contact my hosting provider to get SMTP ports unblocked.

### How to vote
Please give us a thumbs up or thumbs down as a reaction to help us prioritize this feature. Feel free to comment if you have a particular need or comment on how this feature should work.

**Comment by robotdan** [comment]
Related to https://github.com/FusionAuth/fusionauth-issues/issues/310. 

Thanks @LukeHallett we've discussed this option in the past. The primary reason we have not done this yet is due to the vast number of providers. We could make an investment in SendGrid, Twilio, etc - and that would still not cover everyone's use case. 

SMTP is great for this reason because it is a common transport that most providers support. In most production environments any network connectivity issue can usually be resolved. 

We have a similar issue with SMS, we chose to integrate years ago with Twilio for SMS deliver and we have found especially in other parts of the world there are many regional providers. 

The way we plan to solve this for SMS and Email is to support a webhook that would allow you to consume email events and then pass them along to any service or vendor you choose.
